app/controllers/api/conversion_hosts_controller.rb

```diff
@@ -28,9 +22,18 @@ class ConversionHostsController < BaseController
     def create_resource(type, id, data)
       raise BadRequestError, "resource_id must be specified" unless data['resource_id']
       raise BadRequestError, "resource_type must be specified" unless data['resource_type']
-      raise BadRequestError, "invalid resource_type #{data['resource_type']}" unless VALID_TYPES[data['resource_type']]
 
-      resource_type = VALID_TYPES[data['resource_type']]
+      # The scary constantize call below is mitigated by the fact that it won't get
+      # past the following checks we make before passing along the params.
+
+      resource_type = data['resource_type'].classify.safe_constantize
+
+      raise BadRequestError, "invalid resource_type #{data['resource_type']}" unless resource_type
+
+      unless resource_type.respond_to?(:supports_conversion_host?) && resource_type.supports_conversion_host?
```

Comment:
Do we want to show the unsupported_reason when it respond_to, but returns false?

Reply:
Not a show-stopper for merge.